Sources/Lifetime.swift

```diff
@@ -4,37 +4,46 @@ import enum Result.NoError
 /// Represents the lifetime of an object, and provides a hook to observe when
 /// the object deinitializes.
 public final class Lifetime {
-	// MARK: Type properties and methods
-
-	/// Factory method for creating a `Lifetime` and its associated `Token`.
-	///
-	/// - returns: A `(lifetime, token)` tuple.
-	public static func make() -> (lifetime: Lifetime, token: Token) {
-		let token = Token()
-		return (Lifetime(token), token)
+	internal enum Backing {
+		case token(ended: Signal<(), NoError>)
+		case disposable(CompositeDisposable)
```

Comment:
Do we need to support two different backings?

It's only used in one spot and it could be emulated by holding a reference to the token in an `ActionDisposable`.